codex-rs/mcp-server/src/mcp_protocol.rs

```diff
@@ -0,0 +1,1062 @@
+pub struct ConversationSendMessageAccepted {
+    pub accepted: bool,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct ConversationsListResult {
+    pub conversations: Vec<ConversationSummary>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub next_cursor: Option<String>,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct ConversationSummary {
+    pub conversation_id: ConversationId,
+    pub title: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(tag = "type", rename_all = "camelCase")]
+pub enum ToolCallResponseContent {
+    Text { text: String },
+}
+
+// Notifications
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(tag = "method", content = "params", rename_all = "camelCase")]
+pub enum ConversationNotificationParams {
+    InitialState(InitialStateNotificationParams),
+    // sent when a second client connects to the same conversation
+    ConnectionRevoked(ConnectionRevokedNotificationParams),
+    CodexEvent(CodexEventNotificationParams),
+    Cancelled(CancelNotificationParams),
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NotificationMeta {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub conversation_id: Option<ConversationId>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub request_id: Option<RequestId>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct InitialStateNotificationParams {
+    #[serde(rename = "_meta", skip_serializing_if = "Option::is_none")]
+    pub meta: Option<NotificationMeta>,
+    pub initial_state: InitialStatePayload,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct InitialStatePayload {
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub events: Vec<CodexEventNotificationParams>,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct ConnectionRevokedNotificationParams {
+    #[serde(rename = "_meta", skip_serializing_if = "Option::is_none")]
+    pub meta: Option<NotificationMeta>,
+    pub reason: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CodexEventNotificationParams {
+    #[serde(rename = "_meta", skip_serializing_if = "Option::is_none")]
+    pub meta: Option<NotificationMeta>,
+    pub msg: EventMsg,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CancelNotificationParams {
+    pub request_id: RequestId,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub reason: Option<String>,
+}
+
+#[derive(Debug, Clone)]
+pub enum NotificationEnvelope {
+    InitialState(InitialStateNotificationParams),
+    ConnectionRevoked(ConnectionRevokedNotificationParams),
+    Cancelled(CancelNotificationParams),
+    CodexEvent(CodexEventNotificationParams),
+}
+
+impl From<ConversationNotificationParams> for NotificationEnvelope {
```

Comment:
Why is `ConversationNotificationParams` prefixed with `Conversation` but `NotificationEnvelope` isn't?